### ISSUE TITLE
tag setting / getting in nvml check

### ISSUE THREAD

**Issue by blamo-b** [DataDog/integrations-extras#1585]
It seems like there’s a mismatch in the key used to set / get tags in the `known_tags` dict of the nvml integration.

For getting tags, the uuid of the device -- something like `GPU-b922e333-3cd2-3fed-b455-449a8718a546` -- is used as the key.
https://github.com/DataDog/integrations-extras/blob/master/nvml/datadog_checks/nvml/nvml.py#L115
https://github.com/DataDog/integrations-extras/blob/master/nvml/datadog_checks/nvml/nvml.py#L198

For setting tags, in the `refresh_tags` method,  `device.device_ids` (returned from `stub.List(ListPodResourcesRequest())`) is a string like `nvidia1` and used as the key.
https://github.com/DataDog/integrations-extras/blob/master/nvml/datadog_checks/nvml/nvml.py#L235

In short, since `nvidia1` is the key for setting and `GPU-b922a033-3cd2-3fed-b455-449a8718a646` for getting, it doesn’t seem like the tags set by `refresh_tags` would be picked up by the `gather` method

**Additional environment details (Operating System, Cloud provider, etc):**
GCP / GKE 1.22
Agent 7.39.2

**Comment by FlorentClarret** [comment]
Hi! Thanks for opening this issue. 

Pinging @cep21  as maintainer of the nvml integration.

**Comment by alopezz** [comment]
This issue is a bit old, but @cep21 would it be possible for you to have a look at this at some point? Otherwise, @blamo-b, if you still care about this issue, since you seem to have a good grasp of what changes would be needed code-wise, feel free to go ahead and submit a PR with a fix.

**Comment by cep21** [comment]
This plugin was written before datadog had official GPU support.  I believe they now want us to do this: https://www.datadoghq.com/blog/monitor-nvidia-gpus-with-datadog/

We've migrated and aren't using the plugin anymore.
